PR DETAILS USER

elabcius5 opened the pull request:
For this PR, we created a classes for brings user information service

elabcius5 committed:
ADD NEW METHOD

Added new method for shows subscribers list

elabcius5 committed:
ADAPTER

- Tweaked to the subscriber adapter to use SubscriberItem class
- Fixed package structure of SubscribersList to separate each layout of
MVP

elabcius5 committed:
ADDED USER MODEL AND FRAGMENT LAYOUT

- I modified a package structure according a Edwin's indications
- I added a User model class.
- I created a new fragment for shows user information
a. Avatar
b. Full Name
c. Username
d. Company
e. Location
f. Followers
g. Following Counter
h. Repos Counter
i. list of the Repo Names.
- I created a new row for contains a each repository information

elabcius5 committed:
CREATED STRUCTURE FOR SUBSCRIBERSDETAIL

I created a new clases for subscribersdetail like subcriberslist package

elabcius5 committed:
CREATED A FRAGMENT CLASS

I created a Fragment class that contains a details of subriber seleted

elabcius5 committed:
IMPLEMENTED METHOD IN SUBSCRIBER SERVICE

- I implemented a method for shows user information.
- I implemented all classes for consumes a service method

elabcius5 committed:
IMPLEMENTED SUGGESTIONS OF PAST REVISIONS

elabcius5 committed:
ADDED CLASSES FOR SHOWS USER REPOSITORIES

I implemented classes for shows user repositories in a fragment user

elabcius5 committed:
IMPLEMENTED REPO ADAPTER AND REPO ITEM

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/interactor/SubscriberDetailInteractor.java:
```diff
@@ -0,0 +1,26 @@
+        void onFailureRepository(String errorMessage);
+    }
+
+    void getUserData(OnFinishedListener listener, String userName);
```
We used to pass 1st the data, then the listener.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/interactor/SubscriberDetailInteractor.java:
```diff
@@ -0,0 +1,26 @@
+
+    void getUserData(OnFinishedListener listener, String userName);
+
+    void getUserRepositoryData(OnFinishedListener listener, String userName);
```
Same here.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    @Override
+    public void onResume() {
```
Remove unused methods.

juliocatano replied:
Since it is an interface and must be implemented I say, put a `//no-op` comment. Means **no operation**

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -0,0 +1,85 @@
+                             Bundle savedInstanceState) {
+        View viewFragment = inflater.inflate(R.layout.fragment_subscribers_detail, container, false);
+        presenter = new SubscriberDetailPresenterImpl(this);
+        presenter.getUser("keyboardsurfer");
```
Hardcoded string! This need to be dynamic.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailItem.java:
```diff
@@ -0,0 +1,45 @@
+ * @author juan.herrera
+ * @since 31/08/2016
+ */
+public class SubscriberDetailItem extends RelativeLayout {
```
As previous PR comment, this must be a ViewHolder class.

juliocatano replied:
+1

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/commons/models/Repository.java:
```diff
@@ -0,0 +1,490 @@
+    private boolean fork;
+    private String url;
+    @SerializedName("forks_url")
+    private String forksUrl;
```
Do you actually need all this fields?

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscriberItem.java:
```diff
@@ -0,0 +1,48 @@
+ * @author edwin.cobos
+ * @since 30/08/2016
+ */
+public class SubscriberItem extends RelativeLayout {
```
This is a ViewHolder class, why now is a separated class?

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersAdapter.java:
```diff
@@ -18,20 +19,31 @@
  * Adapter class that puts data of subscribers list provided by a service
  *
  * @author juan.herrera
+<<<<<<< HEAD
```
This need to be merged preperly.

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/commons/models/User.java:
```diff
@@ -0,0 +1,218 @@
+    @SerializedName("following_url")
+    String followingUrl;
+    @SerializedName("gists_url")
+    String gistsUrl;
```
do you need all this fields!

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailAdapter.java:
```diff
@@ -0,0 +1,63 @@
+ * @author juan.herrera
+ * @since 31/08/2016
+ */
+public class SubscriberDetailAdapter extends RecyclerView.Adapter<SubscriberDetailAdapter.SubscriberDetailViewHolder> {
```
I would rather name it RepositoryAdapter since it's the purpose of this adapter

daniellemariani commented on app/src/main/res/layout/fragment_subscribers_detail.xml:
```diff
@@ -0,0 +1,66 @@
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentTop="true"
+        android:layout_marginRight="@dimen/img_view_margin"/>
+    <TextView
```
Separate components using a new line

elabcius5 committed:
Merge branch 'master' into DetailsUser
This commit solve many issues and comments

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/commons/models/User.java:
```diff
@@ -0,0 +1,79 @@
+ * @since 30/08/2016
+ */
+public class User {
+    String login;
```
Add private access modifier to all fields/attributes.

lucaslabs commented:
LGreat (minor comment)

edwin-pixel-1 committed:
Added private access

- Added private access to all attributes of User model

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Override
+    public void onDestroy() {
```
You should actually use this method when the fragment is destroyed. Is not being used currently.

edwin-pixel-1 committed:
Override onDestroy method

- Override onDestroy method in SubscriberDetailFragment class to call
  onDestroy method in the presenter.

juliocatano commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,5 +1,6 @@
 <resources>
     <string name="app_name">Github Subscribers</string>
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Delete this string. You are not using it in this PR.

edwin-pixel-1 committed:
REMOVED STRING HELLO WORLD

- Removed default "hello world" string in strings.xml

juliocatano commented:
LG
